lib/board.rb

```diff
@@ -0,0 +1,73 @@
+
+class Board
+
+	attr_accessor :cells
+
+	def initialize
+		@cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+		reset!
+	end
+
+  # Resets the board to empty cells
+	def reset!
+		@cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+	end
+
+  # Display the current board
+	def display
+		puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+		puts "-----------"
+		puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+		puts "-----------"
+		puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+	end
+
+  # Returns the current value of a cell on the board
+	def position(input)
+		@cells[input.to_i - 1]
+	end
+
+  # Is the board full - are there any unoccupied positions
+	def full?
+		!@cells.include?(" " || "")
+	end
+
+  # How many turns have been taken
+	def turn_count
+		count = 0
```

Comment:
Consider using the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```